### ISSUE TITLE
Table is empty

### ISSUE THREAD

**Issue by hotgeart** [kdion4891/valiant#2]
Hi,

My table in the index view is empty. I don't know why.

```php
<div class="card-body">
    {{ dd($html->table()) }}
</div>
```

return this empty table:

![](https://i.imgur.com/G6dA07q.png)

Yet I have data:

```php
public function getIndex(Request $request)
{
    $table = $this->model->table();
    $this->setGoBackUrl();
    dd($table); // <------------
    return $request->ajax()
        ? $table->json
        : view('valiant::models.index', ['model' => $this->model, 'html' => $table->html]);
}
```
![](https://i.imgur.com/gSfKUZx.png)

PS: I don't have any JS error.

**Comment by kdion4891** [comment]
Please post your model code. Are you using a custom view? You also need to push the HTML scripts. 

Also check your browser console / network tab for errors.

**Comment by kdion4891** [comment]
See how the default index view works:

https://github.com/kdion4891/valiant/blob/master/resources/views/models/index.blade.php

**Comment by hotgeart** [comment]
No, I putt my `dd();` in your vendor file to see if the data was there.


```php
<?php

namespace App;

use Illuminate\Database\Eloquent\Model;
use Kdion4891\Valiant\Field;
use Kdion4891\Valiant\Traits\Models\ValiantModel;

class News extends Model
{
    use ValiantModel;

    public function fields()
    {
        return [
            Field::make('ID')
                ->table()->tableSearchSort()->tableDefaultOrder('desc')
                ->detail(),

            Field::make('Title')
                ->table()->tableSearchSort()
                ->detail()
                ->input()->inputCreateEdit()
                ->rulesCreateEdit(['title' => 'required']),

            Field::make('Body')
                ->detail()
                ->inputTextarea()->inputCreateEdit(),

            Field::make('Media')
                ->table()
                ->detail()
                ->inputFile()->inputCreateEdit(),

            Field::make('Published')
                ->table()
                ->detail()
                ->inputCheckbox()->inputCreateEdit(),

            Field::make('Created At')->detail(),
            Field::make('Updated At')->detail(),
        ];
    }
}
```

I've no error in the console

**Comment by kdion4891** [comment]
It's working for me. Here's what I did.

Ran the make command:

	php artisan valiant:make News

Updated the model with `$casts` and the `fields()`:

	class News extends Model
	{
	    use ValiantModel;

	    protected $casts = ['media' => 'array'];

	    public function fields()
	    {
	        return [
	            Field::make('ID')
	                ->table()->tableSearchSort()->tableDefaultOrder('desc')
	                ->detail(),

	            Field::make('Title')
	                ->table()->tableSearchSort()
	                ->detail()
	                ->input()->inputCreateEdit()
	                ->rulesCreateEdit(['title' => 'required']),

	            Field::make('Body')
	                ->detail()
	                ->inputTextarea()->inputCreateEdit(),

	            Field::make('Media')
	                ->table()
	                ->detail()
	                ->inputFile()->inputCreateEdit(),

	            Field::make('Published')
	                ->table()
	                ->detail()
	                ->inputCheckbox()->inputCreateEdit(),

	            Field::make('Created At')->detail(),
	            Field::make('Updated At')->detail(),
	        ];
	    }
	}

You need to cast file fields to `array`. You also need to set your `filesystems` default disk to `public` and symlink it via `php artisan storage:link`.

Then I updated the migration with the field columns:

    public function up()
    {
        Schema::create('news', function (Blueprint $table) {
            $table->bigIncrements('id');
            $table->string('title');
            $table->text('body')->nullable();
            $table->text('media')->nullable();
            $table->boolean('published');
            $table->timestamps();
        });
    }

Ran the migrations:

	php artisan migrate

Then it allowed me to create, edit, and view news just fine. It also worked when I uploaded a `media` file.

One thing I noticed was the ID of your table seems wrong. The table ID should be `table_news`, but yours is `tablehttp:_blueprint.test_news`, have you configured the correct `app_url` in your `.env` file?

You can see how the table ID is determined in the `ValiantModel` trait:

    $html->setTableId('table' . preg_replace('/^[^a-z]+|[^\w:.-]+/i', '_', str_replace(config('app.url'), '', request()->url())));

It takes the current url, removes the `config app.url`, and removes unallowed html ID attribute characters. I think this is why your table is not rendering. Make sure you fix your `app.url` in your `.env` file. I might need to make changes to this logic so it doesn't rely on the config `app.url`.

Let me know if this helps.

**Comment by hotgeart** [comment]
I confirm it's the `APP_URL`. Maybe you should specify it in the getting started section? I know its a bad practice from me, but when I test a package I only update the `DB_DATABASE` and follow the getting started.

**Comment by kdion4891** [comment]
Glad you got it sorted. It says to config your app name, url, database, and mail in the installation guide in the readme and the docs: 

> Configure your .env file with your app name, URL, database, & mail server.

Perhaps I should make this stand out more?

**Comment by hotgeart** [comment]
Ah my bad 🙈. Strange my brain didn't see it haha. May be because its a generic sentence.

Yeah a "⚠" to say the script need the real app_url. The table need a uniq id? Why not just the model name? Or you can dynamically retrieve the domain name. Idk

**Comment by kdion4891** [comment]
Tables need a unique ID per page because tables save their state so when the user goes back to the page it continues where they left off. It has to be unique for datatables logic to remember it correctly.

I'm going to update the readme + docs to be more clear about `.env` file config.

**Comment by kdion4891** [comment]
This bug is now fixed in version 1.0.1. The table ID's no longer rely on the config app URL.
